src/sender_queue/message.rs

```diff
@@ -0,0 +1,78 @@
+use rand::{Rand, Rng};
+
+use Epoched;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum MessageContent<M> {
+    EpochStarted,
+    Algo(M),
+}
+
+impl<M> Rand for MessageContent<M>
+where
+    M: Epoched + Rand,
+{
+    fn rand<R: Rng>(rng: &mut R) -> Self {
+        let message_type = *rng.choose(&["epoch", "algo"]).unwrap();
+
+        match message_type {
+            "epoch" => MessageContent::EpochStarted,
+            "algo" => MessageContent::Algo(rng.gen::<M>()),
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl<M> MessageContent<M>
+where
+    M: Epoched,
+{
+    pub fn with_epoch(self, epoch: <M as Epoched>::Epoch) -> Message<M> {
+        Message {
+            epoch,
+            content: self,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Message<M>
+where
+    M: Epoched,
+{
+    pub epoch: <M as Epoched>::Epoch,
```

Comment:
Isn't that redundant? Could we just rename `MessageContent` to `Message`, and give `EpochStarted` and `Epoch` argument?